src/main/scala/com/databricks/spark/csv/util/TypeCast.scala

```diff
@@ -44,8 +48,10 @@ object TypeCast {
         case _: ShortType => datum.toShort
         case _: IntegerType => datum.toInt
         case _: LongType => datum.toLong
-        case _: FloatType => datum.toFloat
-        case _: DoubleType => datum.toDouble
+        case _: FloatType => Try(datum.toFloat)
+          .getOrElse(java.text.NumberFormat.getInstance(locale).parse(datum).floatValue())
```

Comment:
I think we can't, Users must have a choice for parse double and floats with dots or commas.  The most appropriate way is passing a parameter and the code will be clear. Do you like this option?

Other option is use the locale capturing the exception.. obviously there is a small perfomance penalty.